packages/editor/src/components/profile/Profile.tsx

```
@@ -0,0 +1,245 @@
+  }
+
+  const handleRemoveOAuthServiceClick = (e) => {
+    AuthService.removeUserOAuth(e.currentTarget.id)
+  }
+
+  const enableSocial =
+    authState?.value?.discord ||
+    authState?.value?.facebook ||
+    authState?.value?.github ||
+    authState?.value?.google ||
+    authState?.value?.linkedin ||
+    authState?.value?.twitter
+
+  const handleLogout = async () => {
+    PopupMenuServices.showPopupMenu(UserMenus.Profile)
+    await AuthService.logoutUser()
+    oauthConnectedState.set(Object.assign({}, initialOAuthConnectedState))
+  }
+
+  const showKeys = useHookstate({
+    apiKey: false,
+    userId: false,
+    inviteCode: false
+  })
+
+  const toggleKey = (type: string) => {
+    switch (type) {
+      case 'apiKey':
+        showKeys.set({ ...showKeys.value, apiKey: !showKeys.apiKey.value })
+        break
+      case 'userId':
+        showKeys.set({ ...showKeys.value, userId: !showKeys.userId.value })
+        break
+      case 'inviteCode':
+        showKeys.set({ ...showKeys.value, inviteCode: !showKeys.inviteCode.value })
+        break
+    }
+  }
+
+  return (
+    <>
+      <div className="w-80 bg-neutral-900">
+        <div className="flex gap-2 px-4 py-2">
+          <div className="h-16 w-16 gap-2 overflow-hidden rounded-full bg-slate-300">
+            <img src={avatar.thumbnailResource.url.value} className="h-full w-full object-contain" />
+          </div>
+          <span className="px-2 py-3 text-lg text-white">{name.value}</span>
+        </div>
+        <hr className="mx-4 text-[#A0A1A2] opacity-10" />
+        <div className="px-4 py-2">
+          <div className="grid grid-cols-1 gap-3">
+            <InputGroup
+              className="mr-0 flex flex-row items-center justify-start"
+              label="Invite Code"
+              labelClassName="text-left w-20"
+            >
+              <StringInput
+                type={showKeys.inviteCode.value ? 'text' : 'password'}
```

Did someone request that these fields be hidden by default and then toggleable? We don't hide the invite code or userId anywhere else.